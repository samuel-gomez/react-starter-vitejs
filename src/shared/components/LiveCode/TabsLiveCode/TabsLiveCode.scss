@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.af-tabs {
  &--tabs-live-code {
    .af-tabs__pane {
      padding: 0;
      position: relative;
    }
    .af-tabs__link {
      font-weight: 500;
    }
  }
}

@include media-breakpoint-down(sm) {
  .af-tabs {
    &--tabs-live-code {
      .af-tabs__item--has-icon-left {
        padding-left: 0;
      }
      .af-tabs__link {
        font-size: 0;
        padding: 0 2rem;
      }
      .af-tabs__item .glyphicon {
        font-size: 1rem;
      }
    }
  }
}
