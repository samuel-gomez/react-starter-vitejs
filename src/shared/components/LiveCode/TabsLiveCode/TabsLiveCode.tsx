import { Tabs } from '@axa-fr/react-toolkit-all';
import TitleTabsLiveCode from './TitleTabsLiveCode';
import './TabsLiveCode.scss';

type TTabsLiveCode = {
  children: JSX.Element[];
};

const TabsLiveCode = ({ children }: TTabsLiveCode) => (
  <Tabs classModifier="tabs-live-code" activeIndex="0">
    {children
      .filter(({ props: { hideComponent } }) => !hideComponent)
      .map((component: JSX.Element) => (
        <Tabs.Tab
          key={component.key}
          title={<TitleTabsLiveCode title={component.props.title} icon={component.props.icon} />}
          classModifier="has-icon-left"
        >
          {component}
        </Tabs.Tab>
      ))}
  </Tabs>
);

export default TabsLiveCode;
