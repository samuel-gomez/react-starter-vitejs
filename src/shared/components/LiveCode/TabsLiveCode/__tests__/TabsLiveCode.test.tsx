import { render } from '@testing-library/react';
import TabsLiveCode from '../TabsLiveCode';

describe('<TabsLiveCode />', () => {
  it('Render <TabsLiveCode /> with children', () => {
    const { asFragment } = render(
      <TabsLiveCode>
        <div>test</div>
        <div>test</div>
      </TabsLiveCode>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
