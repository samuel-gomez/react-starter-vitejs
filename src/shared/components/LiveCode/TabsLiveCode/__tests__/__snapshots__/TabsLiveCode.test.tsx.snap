// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TabsLiveCode /> > Render <TabsLiveCode /> with children 1`] = `
<DocumentFragment>
  <div
    class="af-tabs af-tabs--tabs-live-code"
  >
    <ul
      class="af-tabs__control"
    >
      <li
        class="af-tabs__item af-tabs__item--has-icon-left af-tabs__item--active"
      >
        <button
          class="af-tabs__link"
          id="0"
          type="button"
        >
          <i
            class="glyphicon glyphicon-undefined"
          />
           
        </button>
      </li>
      <li
        class="af-tabs__item af-tabs__item--has-icon-left"
      >
        <button
          class="af-tabs__link"
          id="1"
          type="button"
        >
          <i
            class="glyphicon glyphicon-undefined"
          />
           
        </button>
      </li>
    </ul>
    <div
      class="af-tabs__content"
    >
      <div
        class="af-tabs__pane"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
